views.py

```diff
@@ -51,6 +53,12 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
             response.content = "<h1>" + content_msg + "</h1>"
             return response
 
+    if res.resource_type == "CompositeResource" and not path.endswith(".zip"):
+        for f in ResourceFile.objects.filter(object_id=res.id):
+            if path == f.storage_path:
+                if f.has_logical_file and f.logical_file.is_single_file_aggregation:
+                    is_sf_agg_file = True
```

Comment:
@sblack-usu Two if conditions can be combined into one, and a ```break``` statement can be added after setting ```is_sf_agg_file``` to True to break out the for loop.